### PR TITLE
Add API for getting logged data associated to a Model

### PR DESCRIPTION
**Pull request by aleksmitov**
Add API for getting logged data associated to a Model.
Refactored some code to avoid circular dependencies.

Fixes https://github.com/layerai/roadmap/issues/1

### CODE REVIEW

File: layer/contracts/logged_data.py [hunk 1]
```diff
@@ -395,3 +398,92 @@ class Markdown:
...
+
+class LoggedDataObject:
+    def __init__(self, logged_data: LoggedData, epoch: Optional[int] = None):
+        self._logged_data = logged_data
+        self._epoch = epoch
+
+    @staticmethod
+    def _download_object_to(url: str, file_path: Path) -> None:
+        response = requests.get(url)
+        with open(file_path, "wb") as filer_handler:
```
Comment (cihati) [inline layer/contracts/logged_data.py:411]: nit: `file_handler`

File: layer/contracts/models.py [hunk 2]
```diff
@@ -111,3 +114,15 @@ def set_model_runtime_objects(
     ) -> "Model":
         self._model_runtime_objects = model_runtime_objects
         return self
+
+    def add_log_data_runner(self, log_data_runner: LogDataRunner) -> None:
+        self._logged_data_runner = log_data_runner
+
+    def get_metadata(self, tag: str, step: Optional[int] = None) -> LoggedDataObject:
```
Comment (cihati) [inline layer/contracts/models.py:121]: Is this only meant for models, or datasets too? cc @volkangurel

File: layer/contracts/logged_data.py [hunk 3]
```diff
@@ -395,3 +398,92 @@ class Markdown:
...
+            )
+            with tempfile.NamedTemporaryFile() as temp_file:
+                self._download_object_to(url, Path(temp_file.name))
+                image = pil_open(temp_file)
+                image.load()
+                return image
+        elif self.is_markdown():
+            return self._logged_data.data
+        else:
+            raise Exception(
```
Comment (cihati) [inline layer/contracts/logged_data.py:442]: Maybe have a test for this behavior (same for `download_to`'s such exception too)?

**Review by cihati — COMMENTED** [review]
Good work!

File: layer/contracts/models.py (hunk 2 above)
Comment (volkangurel) [reply layer/contracts/models.py:121]: This is for datasets too.

**Comment by aleksmitov** [comment]
Fixes https://github.com/layerai/roadmap/issues/1

**Review by cihati — APPROVED** [review]
let's do datasets separately.

**Pull request merged**